src/integrations/admin/indexation-integration.php

```diff
@@ -149,47 +170,19 @@ public function enqueue_scripts() {
 			return;
 		}
 
-		\add_action( 'admin_footer', [ $this, 'render_indexation_modal' ], 20 );
+		$this->indexation_action_is_link = ! $this->yoast_tools_page_conditional->is_met();
+
 		if ( $this->is_indexation_warning_hidden() === false ) {
 			\add_action( 'admin_notices', [ $this, 'render_indexation_warning' ], 10 );
 		}
 		elseif ( $this->term_indexation->get_total_term_permalinks_null() > $shutdown_limit ) {
 			\add_action( 'admin_notices', [ $this, 'render_indexation_permalink_warning' ], 10 );
 		}
 
-		$this->asset_manager->enqueue_script( 'indexation' );
-		$this->asset_manager->enqueue_style( 'admin-css' );
-
-		$data = [
-			'amount'  => $this->get_total_unindexed(),
-			'ids'     => [
-				'count'    => '#yoast-indexation-current-count',
-				'progress' => '#yoast-indexation-progress-bar',
-			],
-			'restApi' => [
-				'root'      => \esc_url_raw( \rest_url() ),
-				'endpoints' => [
-					'prepare'  => Indexable_Indexation_Route::FULL_PREPARE_ROUTE,
-					'posts'    => Indexable_Indexation_Route::FULL_POSTS_ROUTE,
-					'terms'    => Indexable_Indexation_Route::FULL_TERMS_ROUTE,
-					'archives' => Indexable_Indexation_Route::FULL_POST_TYPE_ARCHIVES_ROUTE,
-					'general'  => Indexable_Indexation_Route::FULL_GENERAL_ROUTE,
-					'complete' => Indexable_Indexation_Route::FULL_COMPLETE_ROUTE,
-				],
-				'nonce'     => \wp_create_nonce( 'wp_rest' ),
-			],
-			'message' => [
-				'indexingCompleted' => '<span class="wpseo-checkmark-ok-icon"></span>' . \esc_html__( 'Good job! You\'ve sped up your site.', 'wordpress-seo' ),
-				'indexingFailed'    => \__( 'Something went wrong while optimizing the SEO data of your site. Please try again later.', 'wordpress-seo' ),
-			],
-			'l10n'    => [
-				'calculationInProgress' => \__( 'Optimization in progress...', 'wordpress-seo' ),
-				'calculationCompleted'  => \__( 'Optimization completed.', 'wordpress-seo' ),
-				'calculationFailed'     => \__( 'Optimization failed, please try again later.', 'wordpress-seo' ),
-			],
-		];
-
-		\wp_localize_script( WPSEO_Admin_Asset_Manager::PREFIX . 'indexation', 'yoastIndexationData', $data );
+		// Only enqueue indexation assets when the action is a button.
+		if ( $this->indexation_action_is_link === false ) {
```

Comment:
I would use the conditional at this point. This makes it more clear that this is only needed on the tools page :)
```suggestion
		if ( $this->yoast_tools_page_conditional->is_met() ) {
```

At the moment this variable is describing the outcome instead of the reason of that outcome. If there is need to change the link by button, you have to rename the variable now.

You can also move the conditional itself to the `enqueue_indexation_assets` by doing a early return when it is not the tools page.

Reply:
I want to add that it is more clear now that this only happens on the tools page :)

Reply:
Reworked this into a string option (as discussed). It is a lot more verbose now.
I did not go for the moving of the conditional. It does not make sense to me that if you call that function nothing could happen. This keeps it a cleaner function.
Also did not move it to the constructor because that would introduce side-effects (calling a function that does things). I toyed with the idea to do it in the register hooks but that muddies the waters too. So kept it in the enqueue function now.